crates/protocol/src/fee.rs

```diff
@@ -50,6 +50,27 @@ fn tx_estimated_size_fjord(input: &[u8]) -> U256 {
         .max(U256::from(100_000_000))
 }
 
+/// Calculate the gas cost of a transaction based on L1 block data posted on L2 post-ecotone.
+/// This is a special case where the gas cost function uses the bedrock gas cost function,
+/// but the data gas is calculated using the ecotone data gas function.
+pub fn calculate_tx_l1_cost_bedrock_empty_scalars(
+    input: &[u8],
+    l1_fee_overhead: U256,
+    base_fee: U256,
+    l1_fee_scalar: U256,
+) -> U256 {
+    if input.is_empty() || input.first() == Some(&0x7F) {
+        return U256::ZERO;
+    }
+
+    let rollup_data_gas_cost = data_gas_regolith(input);
+    rollup_data_gas_cost
+        .saturating_add(l1_fee_overhead)
+        .saturating_mul(base_fee)
+        .saturating_mul(l1_fee_scalar)
+        .wrapping_div(U256::from(1_000_000))
```

Comment:
What is this magic number? Possibly out of scope of this pr to replace with a constant

Reply:
Believe it's just a normalization value for units.